fv3core/stencils/updatedzc.py

```diff
@@ -220,44 +78,29 @@ def update_dz_c(
             xfx = p_weighted_average_bottom(ut, dp_ref)
             yfx = p_weighted_average_bottom(vt, dp_ref)
     with computation(PARALLEL), interval(...):
+        gz_x = gz
+        gz_x = corners.fill_corners_2cells_mult_x(gz_x, gz_x, 1.0, 1.0, 1.0, 1.0)
+        gz_y = gz_x
+        gz_y = corners.fill_corners_2cells_mult_y(gz_y, gz_y, 1.0, 1.0, 1.0, 1.0)
         fx, fy = xy_flux(gz_x, gz_y, xfx, yfx)
-        # TODO: check if below gz is ok, or if we need gz_y to pass this
-        gz = (gz_y * area + fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) / (
-            area + xfx - xfx[1, 0, 0] + yfx - yfx[0, 1, 0]
-        )
+        # TODO: region for local validation only
+        with horizontal(
+            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+        ):
+            gz = (gz_y * area + fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) / (
+                area + xfx - xfx[1, 0, 0] + yfx - yfx[0, 1, 0]
+            )
     with computation(PARALLEL), interval(-1, None):
-        rdt = 1.0 / dt
-        ws3 = (zs - gz) * rdt
+        rdt = 1.0 / dt2
+        # TODO: region for local validation only
+        with horizontal(
+            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+        ):
+            ws = (z_surface - gz) * rdt
```

Comment:
`rdt` only gets used in this one place below, so you could condense this for readability (pending possibly updating the `dt2` name).

```suggestion
        # TODO: region for local validation only
        with horizontal(
            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
        ):
            ws = (z_surface - gz) / dt2
```